services/endpointsrv/service.go

```diff
@@ -460,3 +482,49 @@ func (e *EndpointService) EndpointCheckStatus(ctx context.Context, checkID strin
 		Steps:   out,
 	}, nil
 }
+
+func (e *EndpointService) EndpointVersionCreate(
+	ctx context.Context,
+	projectID,
+	parentEndpointID,
+	execID string,
+	promote bool,
+) (types.EndpointVersion, error) {
+	versionID := typeid.Must(typeid.New("version")).String()
+
+	log.Debug().
+		Str("project", projectID).
+		Str("parent_endpoint", parentEndpointID).
+		Str("exec_id", execID).
+		Str("version", versionID).
+		Bool("promote", promote).
```

Comment:
Could you use the context keys defined [here](https://github.com/unweave/unweave/blob/master/api/types/context.go#L8-L16) so we can reuse them and not change them uniformly across all occurrences.

Reply:
Fixed - thanks for pointing this out 💯